thruster/src/context/typed_hyper_context.rs

```diff
@@ -98,35 +100,15 @@ impl<S: 'static + Send> TypedHyperContext<S> {
     ///
     /// Get the body as a string
     ///
-    pub async fn get_body(self) -> Result<(String, TypedHyperContext<S>), Error> {
-        let ctx = match self.request_body {
-            Some(_) => self,
-            None => self.into_owned_request(),
-        };
-
-        let mut results = "".to_string();
-        let mut unwrapped_body = ctx.request_body.unwrap();
-        while let Some(chunk) = unwrapped_body.next().await {
-            // TODO(trezm): Dollars to donuts this is pretty slow -- could make it faster with a
-            // mutable byte buffer.
-            results = format!("{}{}", results, String::from_utf8_lossy(chunk?.as_ref()));
-        }
-
-        Ok((
-            results,
-            TypedHyperContext {
-                body: ctx.body,
-                query_params: ctx.query_params,
-                headers: ctx.headers,
-                status: ctx.status,
-                hyper_request: ctx.hyper_request,
-                request_body: Some(Body::empty()),
-                request_parts: ctx.request_parts,
-                extra: ctx.extra,
-                http_version: ctx.http_version,
-                cookies: ctx.cookies,
-            },
-        ))
+    pub async fn get_body(&mut self) -> Result<String, Box<dyn std::error::Error>> {
```

Comment:
What about `req_body`?

On a side note, I am starting to see a clear separation of "req" stuff versus "res". 
What about separate then (maybe in another PR) in the context, like `context.request.header("Authorization")` and `context.response.headers.set("Content-Type", "application/json")`/`context.response.set("Content-Type", "application/json")`.

Reply:
I think `req_body` makes sense.

As to the second half, the problem then becomes that we start expanding the number of items required to implement a `Context`. We'd need more traits for `Request`, `Response`, and then a few more methods on the `Context` trait that expose those fields for access like this. 

Koa does a decent job with this that might make sense, which is more of a half-measure -- they have all of the request specific fields in an accessor on the context. Let me do some fussing and see if that's a reasonable life improvement in a different PR.

Reply:
Sure, I think should be reasonable as you say, not too much or too few